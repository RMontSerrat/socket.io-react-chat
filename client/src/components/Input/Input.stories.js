import React from 'react';
import { storiesOf } from '@storybook/react';
import Input from './Input';

storiesOf('Input', module)
  .add('default', () => (
    <div style={{ padding: '20px', backgroundColor: '#025a3f' }}>
      <Input placeholder="Digite aqui..." />
    </div>
  ));
