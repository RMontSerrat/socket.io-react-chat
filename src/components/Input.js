import styled from 'styled-components';

const Input = styled.input`
  background-color: ${props => props.theme.colorWhite};
  padding: 10px;
  width: calc(100% - 20px);
  color: ${props => props.theme.colorPrimary};
  border-radius: ${props => props.theme.borderRadius};
  font-family: ${props => props.theme.fontFamily};
  border: 0;
  outline: none;
`

export default Input;