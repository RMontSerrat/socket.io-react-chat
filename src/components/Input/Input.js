import styled from 'styled-components';

const Input = styled.input`
  background-color: ${props => props.theme.color.white};
  padding: 15px 10px;
  width: calc(100% - 20px);
  color: ${props => props.theme.color.primary};
  border-radius: ${props => props.theme.borderRadius};
  font-family: ${props => props.theme.font.primary};
  border: 0;
  outline: none;
`

export default Input;