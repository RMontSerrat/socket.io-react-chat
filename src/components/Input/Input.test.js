import React from "react";
import { render, cleanup } from "react-testing-library";
import Input from './Input';

afterEach(cleanup);
describe('<Input />', () => {
  it("should render", () => {
    const { asFragment } = render(
      <Input value="romulo" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});