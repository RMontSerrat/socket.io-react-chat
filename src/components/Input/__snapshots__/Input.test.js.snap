// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render 1`] = `
<DocumentFragment>
  .c0 {
  padding: 15px 10px;
  width: calc(100% - 20px);
  border: 0;
  outline: none;
}

<input
    class="c0"
    value="romulo"
  />
</DocumentFragment>
`;
